services/core-web/src/components/noticeOfWork/NOWActionWrapper.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { PropTypes } from "prop-types";
+import { withRouter } from "react-router-dom";
+import CustomPropTypes from "@/customPropTypes";
+import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
+import { getNoticeOfWork } from "@common/selectors/noticeOfWorkSelectors";
+
+/**
+ * @constant NOWActionWrapper conditionally renders NoW actions based on various conditions (ie, Rejected, Permit issued, client delay, stages not started, etc)
+ * persists permissions using authWrapper - These actions are not visible to admin if disabled.
+ */
+
+const propTypes = {
+  noticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
+  history: PropTypes.shape({ location: PropTypes.objectOf(PropTypes.any) }).isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.element.isRequired),
+    PropTypes.element.isRequired,
+  ]).isRequired,
+};
+
+const defaultProps = {};
+// eslint-disable-next-line react/prefer-stateless-function
+export class NOWActionWrapper extends Component {
+  // state = { currentTab: "" };
+
+  // componentDidMount() {
```

Comment:
I assume that later, we conditionally set disabled here based on the combined logic of delayed + processed status?

Reply:
I was planning to use this logic to know if a specific tab should be disabled - ie if consultation has not started yet, then any button on the consultation tab should be hidden until that stage is in progress. There will likely be additional logic added for the progress related disabled-buttons.